modules/imgproc/test/test_thresh.cpp

```diff
@@ -520,6 +520,55 @@ TEST(Imgproc_Threshold, threshold_dryrun)
     }
 }
 
+TEST(Imgproc_Threshold, threshold_mask)
+{
+    std::vector<bool> useROIs = {false, true};
+    std::vector<int> depths = {CV_8U, CV_16U, CV_16U, CV_16S, CV_32F, CV_64F};
+    std::vector<int> channels = {1, 3};
+    std::vector<int> threshTypes = {THRESH_BINARY, THRESH_BINARY_INV, THRESH_TRUNC, THRESH_TOZERO, THRESH_TOZERO_INV};
+    std::vector<int> threshFlags = {0};//{THRESH_OTSU, THRESH_TRIANGLE} there is no way to compare OTSU/TRIANGLE since the threshold will be different
+    for(bool useROI : useROIs)
+    {
+        for(int depth : depths)
+        {
+            for(int cn : channels)
+            {
+                for(int threshType : threshTypes)
+                {
+                    for(int threshFlag : threshFlags)
+                    {
```

Comment:
I propose to use parameterized google test here. It generates all "loops" by itself and print options combination in case of failure